sim.c

```diff
@@ -765,19 +765,21 @@ BEGIN_OPERATOR(zig)
   LOWERCASE_REQUIRES_BANG;
   Glyph* gline = gbuffer + width * y;
   gline[x] = '.';
-  if (x + 1 == width)
-    return;
-  if (gline[x + 1] == '.') {
+  if (gline[x + 1] == '.' && x + 1 != width) {
     gline[x + 1] = This_oper_char;
     mbuffer[width * y + x + 1] |= (U8)Mark_flag_sleep;
   } else {
-    Usz n = 256;
+    Usz n = 512;
```

Comment:
Why does this need to be increased? 256 is a lot. The higher this number is pushes, the more memory prefetching the CPU will perform during speculative execution.